### PR TITLE
Fix Coverity issues

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/asn1/a_digest.c [hunk 1]
```diff
@@ -76,7 +76,7 @@ int asn1_item_digest_with_libctx(const ASN1_ITEM *it, const EVP_MD *md,
 #endif
             fetched_md = EVP_MD_fetch(libctx, EVP_MD_name(md), propq);
     }
-     if (fetched_md == NULL)
+     if (fetched_md == NULL || i < 0)
```
Comment (kroeckx) [inline crypto/asn1/a_digest.c:79]: Please add the check for `i` ASN1_item_i2d() call instead.

This line has a space too much.

Please document this function.

File: crypto/evp/digest.c [hunk 2]
```diff
@@ -335,11 +335,18 @@ int EVP_DigestFinal(EVP_MD_CTX *ctx, unsigned char *md, unsigned int *size)
...
+    size_t mdsize = 0;
 
-    if (ctx->digest == NULL || ctx->digest->prov == NULL)
+    if (ctx->digest == NULL)
+        return 0;
+
+    sz = EVP_MD_size(ctx->digest);
+    if (sz < 0)
+        return 0;
+    mdsize = sz;
```
Comment (kroeckx) [inline crypto/evp/digest.c:348]: I see no reason for the `sz` variable, just assign it to `mdsize`.

EVP_MD_size() is not documented as being able to return an error.

Comment (kroeckx) [reply crypto/evp/digest.c:348]: Oh, EVP_MD_size() of course returns an int, but mdsize is a size_t.

File: providers/implementations/signature/rsa.c [hunk 3]
```diff
@@ -725,7 +725,8 @@ static int rsa_digest_signverify_init(void *vprsactx, const char *mdname,
 {
     PROV_RSA_CTX *prsactx = (PROV_RSA_CTX *)vprsactx;
 
-    prsactx->flag_allow_md = 0;
+    if (prsactx != NULL)
+        prsactx->flag_allow_md = 0;
```
Comment (kroeckx) [inline providers/implementations/signature/rsa.c:729]: The rest of the code seems to assume that `vprsactx != NULL`, like the `prsactx->mdctx = EVP_MD_CTX_new();` line.

File: ssl/ssl_rsa.c [hunk 4]
```diff
@@ -639,7 +639,7 @@ static int use_certificate_chain_file(SSL_CTX *ctx, SSL *ssl, const char *file)
     if (ctx != NULL) {
         passwd_callback = ctx->default_passwd_callback;
         passwd_callback_userdata = ctx->default_passwd_callback_userdata;
-    } else {
+    } else if (ssl != NULL) {
```
Comment (kroeckx) [inline ssl/ssl_rsa.c:642]: I think the code assumes that either you have an ssl, or a ctx, and that if you somehow get here with `ssl == NULL`, something else is wrong.

File: crypto/asn1/a_digest.c (hunk 1 above)
Comment (slontis) [reply crypto/asn1/a_digest.c:79]: Moved the line up... Documentation is outside the scope.

File: crypto/evp/digest.c (hunk 2 above)
Comment (slontis) [reply crypto/evp/digest.c:348]: yep.. There are a few of these.

File: providers/implementations/signature/rsa.c (hunk 3 above)
Comment (slontis) [reply providers/implementations/signature/rsa.c:729]: Except that rsa_signature_init does the NULL check and returns an error.

File: ssl/ssl_rsa.c (hunk 4 above)
Comment (slontis) [reply ssl/ssl_rsa.c:642]: I think it is ok to leave this as a false positive and remove the change.. Thanks.

**Comment by slontis** [comment]
ping

**Review by t8m — APPROVED** [review]
LGTM

**Comment by slontis** [comment]
Thanks. merged to master

**Pull request closed without merging**